test/unit/webgl/p5.RendererGL.js

```diff
@@ -447,6 +447,68 @@ suite('p5.RendererGL', function() {
       assert.deepEqual([0, 0, 0, 255], mixAndReturn(myp5.DARKEST, 255));
       done();
     });
+
+    test('blendModes match 2D mode', function(done) {
+      myp5.createCanvas(10, 10, myp5.WEBGL);
+      myp5.setAttributes({ alpha: true });
+      const ref = myp5.createGraphics(myp5.width, myp5.height);
+      ref.translate(ref.width / 2, ref.height / 2); // Match WebGL mode
+
+      const testBlend = function(target, colorA, colorB, mode) {
+        target.clear();
+        target.push();
+        target.background(colorA);
+        target.blendMode(mode);
+        target.noStroke();
+        target.fill(colorB);
+        target.rectMode(target.CENTER);
+        target.rect(0, 0, target.width, target.height);
+        target.pop();
+        return target.get(0, 0);
+      };
+
+      const assertSameIn2D = function(colorA, colorB, mode) {
+        const refColor = testBlend(myp5, colorA, colorB, mode);
+        const webglColor = testBlend(ref, colorA, colorB, mode);
+        if (refColor[3] === 0) {
+          assert.equal(webglColor[3], 0);
+        } else {
+          assert.deepEqual(
+            refColor,
+            webglColor,
+            `Blending ${colorA} with ${colorB} using ${mode}`
+          );
+        }
+      };
+
+      const red = '#F53';
+      const blue = '#13F';
+      assertSameIn2D(red, blue, myp5.BLEND);
+      assertSameIn2D(red, blue, myp5.ADD);
+      assertSameIn2D(red, blue, myp5.DARKEST);
+      assertSameIn2D(red, blue, myp5.LIGHTEST);
+      assertSameIn2D(red, blue, myp5.EXCLUSION);
+      assertSameIn2D(red, blue, myp5.MULTIPLY);
+      assertSameIn2D(red, blue, myp5.SCREEN);
+      assertSameIn2D(red, blue, myp5.REPLACE);
+      assertSameIn2D(red, blue, myp5.REMOVE);
+      done();
+    });
+
+    test('blendModes are included in push/pop', function(done) {
+      myp5.createCanvas(10, 10, myp5.WEBGL);
+      myp5.blendMode(myp5.MULTIPLY);
+      myp5.push();
+      myp5.blendMode(myp5.ADD);
+      assert.equal(myp5._renderer.curBlendMode, myp5.ADD, 'Changed to ADD');
+      myp5.pop();
+      assert.equal(
+        myp5._renderer.curBlendMode,
+        myp5.MULTIPLY,
+        'Resets to MULTIPLY'
+      );
+      done();
+    });
```

Comment:
This is a great set of unit tests. This is a great safeguard against unexpected divergence between the renderers down the road.